llvm/lib/Analysis/BranchProbabilityInfo.cpp

```diff
@@ -828,12 +828,13 @@ void BranchProbabilityInfo::computeEestimateBlockWeight(
   do {
     while (!LoopWorkList.empty()) {
       const LoopBlock LoopBB = LoopWorkList.pop_back_val();
-
-      if (EstimatedLoopWeight.count(LoopBB.getLoopData()))
+      const LoopData LD = LoopBB.getLoopData();
+      if (EstimatedLoopWeight.count(LD))
         continue;
 
-      SmallVector<BasicBlock *, 4> Exits;
-      getLoopExitBlocks(LoopBB, Exits);
+      if (!LoopExitBlocks.count(LD))
```

Comment:
It would be better to use `find()` here to avoid the duplicate lookup.

Reply:
Sorry, I don't quite get it.
What do you think of using `try_emplace()` ?
```cpp
      auto Res = LoopExitBlocks.try_emplace(LD);
      SmallVectorImpl<BasicBlock *> &Exits = Res.first->second;
      if (Res.second)
        getLoopExitBlocks(LoopBB, Exits);
      auto LoopWeight = getMaxEstimatedEdgeWeight(
          LoopBB, make_range(Exits.begin(), Exits.end()));
```

Reply:
I meant using find() to avoid doing both the count() and then below another lookup using LoopExitBlocks[LD].

Your suggestion of using try_emplace also covers the insertion path, so I guess that's even better.